text/353-cfn-registry-constructs.md

```diff
@@ -0,0 +1,467 @@
+# Constructs for Public CloudFormation Extensions
+
+* **Original Author(s):**: @eladb
+* **Tracking Issue**: #353
+* **API Bar Raiser**: @rix0rrr
+
+A set of construct libraries which includes generated constructs for all CloudFormation 
+resources and modules published to the public CloudFormation Registry.
+
+## README
+
+The `@cdk-cloudformation-extensions/xxx` construct libraries includes generated strongly-typed "L1" constructs 
+for all the public extensions (resources and modules) in the AWS CloudFormation public registry. This library
+makes it easier to use public CloudFormation extensions in your CDK apps.
+
+For example, let's say I want to define a GitHub repository using the `TF::GitHub::Repository` resource:
+
+```ts
+import { Repository } from '@cdk-cloudformation-extensions/tf-github-repository';
+
+new Repository(this, 'MyRepo', {
+  name: 'my-repo',
+  description: 'My awesome project',
+  licenseTemplate: 'apache-2.0',
+});
+```
+
+For each type (e.g. `TF::GitHub::Repository` in the above example) in 
+the public CloudFormation Registry, a module is available under 
+the scope `@cdk-cloudformation-extensions/NAMESPACE-TYPE`. This library
+includes a construct class and all the relevant data types for this
+type.
+
+The module version corresponds to the version of the type schema
+in the public registry.
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+We are publishing a set of construct libraries (in all JSII languages) which include constructs (and auxiliary types) for 
+all the resources and modules in the public CloudFormation registry.
+
+### Why should I use this feature?
+
+This library makes it easier to discover and use CloudFormation extensions in CDK apps.
+
+Previously, in order to use a resource from the CloudFormation registry, you would need to look up the resource documentation
+and use the low-level, weakly-typed `CfnResource` class to define it in your CDK app:
+
+```ts
+new CfnResource(this, 'MyEksCluster', {
+  type: 'AWSQS::EKS::Cluster',
+
+  // weakly-typed!
+  properties: {
+    name: 'my-new-cluster',
+    tags: [ { key: 'foo', value: 'bar' } ],
+  },
+});
+```
+
+With `@cdk-cloudformation-extensions` all public resources and modules will be listed in the **Construct Hub** like any
+other construct and by importing the relevant `cdk-cloudformation-extensions` module into their projects, they will be able to
+use them via strongly-typed classes. IDEs will show type information and inline help derived from the
+extension schema.
+
+```ts
+import { Cluster } from '@cdk-cloudformation-extensions/awsqs-eks';
```

Comment:
This module name is inconsistent with the previous example. To be in line it would have to be

```ts
import { Cluster } from '@cdk-cloudformation-extensions/awsqs-eks-cluster';
```

Which one was intended?

Reply:
The `awssqs-eks-cluster` one. Fixed.